ext/opentelemetry-ext-docker-tests/tests/postgres/test_psycopg_functional.py

```diff
@@ -0,0 +1,130 @@
+        if cls._cursor:
+            cls._cursor.close()
+        if cls._connection:
+            cls._connection.close()
+
+    @fixture(autouse=True)
+    def connect_to_db(self):
+        if self._connection is None:
+            # Try to connect to DB
+            for i in range(5):
+                try:
+                    self._connection = psycopg2.connect(
+                        dbname=POSTGRES_DB_NAME,
+                        user=POSTGRES_USER,
+                        password=POSTGRES_PASSWORD,
+                        host=POSTGRES_HOST,
+                        port=POSTGRES_PORT,
+                    )
+                    self._cursor = self._connection.cursor()
+                    break
+                except Exception as err:
+                    print(err)
+                time.sleep(5)
+            else:
+                raise Exception("Failed to connect to DB")
+
+    def setUp(self):
+        self._span_exporter.clear()
+
+    def validate_spans(self):
+        spans = self._span_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 2)
+        for span in spans:
+            if span.name == "rootSpan":
+                root_span = span
+            else:
+                child_span = span
+            self.assertIsInstance(span.start_time, int)
+            self.assertIsInstance(span.end_time, int)
+        self.assertIsNot(root_span, None)
+        self.assertIsNot(child_span, None)
+        self.assertEqual(root_span.name, "rootSpan")
+        self.assertEqual(child_span.name, "postgresql.opentelemetry-tests")
+        self.assertIsNotNone(child_span.parent)
+        self.assertEqual(child_span.parent.name, root_span.name)
+        self.assertIs(child_span.kind, trace_api.SpanKind.CLIENT)
+        self.assertEqual(
+            child_span.attributes["db.instance"], POSTGRES_DB_NAME
+        )
+        self.assertEqual(child_span.attributes["net.peer.name"], POSTGRES_HOST)
+        self.assertEqual(child_span.attributes["net.peer.port"], POSTGRES_PORT)
+
+    def test_execute(self):
+        """Should create a child span for execute method
+        """
+        with self._tracer.start_as_current_span("rootSpan"):
+            self._cursor.execute(
+                "CREATE TABLE IF NOT EXISTS test (id integer)"
+            )
+            self._connection.commit()
+        self.validate_spans()
+
+    def test_executemany(self):
+        """Should create a child span for executemany
+        """
+        with self._tracer.start_as_current_span("rootSpan"):
+            data = ("1", "2", "3")
+            stmt = "INSERT INTO test (id) VALUES (%s)"
+            self._cursor.executemany(stmt, data)
+        self.validate_spans()
+
+    def test_callproc(self):
+        """Should create a child span for callproc
+        """
+        with self._tracer.start_as_current_span("rootSpan"):
+            try:
+                self._cursor.callproc("test", ())
+            except Exception as err:
```

Comment:
does this exception need to be caught? seems like it's fine to let it raise up.

Even if it's unrelated to our code, exceptions are a separate code path that we should exercise explicitly.

Reply:
Stored procedure doesn't exist so callproc will fail but Span will be created, I want to avoid as much pre setup as possible for this tests

Reply:
You might want an `assertRaises` here instead then, the tests generally shouldn't print anything.